### ISSUE TITLE
Disk quota for institutions' recordings

### ISSUE THREAD

**Issue by mconf-daileon** [mconf-rnp/mconf-web#65]
---

Author Name: **Leonardo Daronco** (Leonardo Daronco)
Original Redmine Issue: 1729, http://dev.mconf.org/redmine/issues/1729

Original Assignee: Lucas Zawacki

---

Add a new option to institutions to configure a disk quota for recordings created in rooms that belong to the institution.
Whenever a meeting is being created, the available quota and the used quota should be checked. If there's no more space available for the target institution, the meeting should be forced to use the flag "record=false" to disable the option to record.

The disk space used by each recording is returned by @getRecordings@, see #1648.
This is an option available on Mconf-Live only, not on BigBlueButton, so it should be considered as optional. If not present, assume the recording has size zero.

Other considerations:
* A quota of zero or @nil@ should mean that it is unlimited.
* Ideally the user should be notified whenever the meeting is forced to not record because of the quota. There's no obvious way to do it, however, so this is just an optional feature to be considered. For admins, however, log messages should be added and they should be enough.
* If calculating the disk space used by all recordings is slow/complicated, this value could be cached in the institutions table. Whenever a recording is added/removed/edited, it can be recalculated. Would be nice to have this so we could also show the % of the quota in use without having to calculate it.

**Comment by mconf-daileon** [comment]
---

Original Redmine Comment
Author Name: **Leonardo Daronco** (Leonardo Daronco)
Original Date: 2015-08-12T16:45:22Z

---

Another ticket was created to include an option to notify users why they can't record: #1736
Once it is done, we should also notify users properly when they can't record because their institution's quota was exceeded.

**Comment by mconf-daileon** [comment]
---

Original Redmine Comment
Author Name: **Leonardo Daronco** (Leonardo Daronco)
Original Date: 2015-09-23T18:13:48Z

---

Issue #1729 has been updated by Lucas Zawacki.

https://github.com/mconf-rnp/mconf-web/pull/15
Still missing notifications, doing them on the other bug page

**Comment by mconf-daileon** [comment]
---

Original Redmine Comment
Author Name: **Leonardo Daronco** (Leonardo Daronco)
Original Date: 2015-09-30T18:12:27Z

---

See https://github.com/mconf-rnp/mconf-web/pull/15
